modules/shared/src/main/scala/org/tessellation/currency/schema/currency.scala

```diff
@@ -105,7 +113,9 @@ object currency {
     stateProof: CurrencySnapshotStateProof,
     epochProgress: EpochProgress,
     dataApplication: Option[DataApplicationPart] = None,
-    version: SnapshotVersion = SnapshotVersion("0.0.1")
+    version: SnapshotVersion = SnapshotVersion("0.0.1"),
+    adminFeeKeyMessage: Option[AdminFeeKeyMessage] = None,
+    stakingMessage: Option[StakingMessage] = None
```

Comment:
We can't really add it right now (schema change). One thing I was pointing out in a discussion earlier is that maybe it should be a generic (or sealed but extendable) `Event` and a state of such events would be persisted in the currencySnapshotInfo. I don't think we need these fields on the snapshot itself as they seem to remain mostly constant right? Anyway we can't merge it yet as this is a plain schema change (which we are not handling right now).

Reply:
I'd agree that instead of rewriting the same data over and over again, it should happen once and be stored for easy access in the offchain data (currency snapshot info). @ryle-goehausen ?

Reply:
I had the same thought and mentioned it to Ryle yesterday. He mentioned we're adding it here because of key rotation; subsequent runs could change those values. Is there a better place for us to put it where we can achieve the same goal?

@ryle-goehausen please correct me if misunderstood our discussion.

Reply:
@sdavidp I don't see why `subsequent runs could change those values` affect the place and how we would store it. Can you elaborate more on that here or on slack?

Reply:
If I understand correctly:

Option 1:
-  snapshot A (stakingMessage: Signed(_abcdef_))
-  snapshot B (stakingMessage: Signed(_abcdef_))
**<-- changing the staking message to _zxcvbn_**
-  snapshot C (stakingMessage: Signed(_zxcvbn_))
-  snapshot D (stakingMessage: Signed(_zxcvbn_))

Option 2:

- snapshot A (events: stakingMessage=Signed(_abcdef_)) `last stakingMessage=Signed(_abcdef_)`
- snapshot B () `last stakingMessage=Signed(_abcdef_)`
**<-- changing the staking message to _zxcvbn_**
- snapshot C (events: stakingMessage=Signed(_zxcvbn_)) `last stakingMessage=Signed(_zxcvbn_)`
- snapshot D () `last stakingMessage=Signed(_zxcvbn_)`

where `lastStakingMessage` can be stored in snapshot companion object (snapshot info or equivalent).

The latter is more generic and more flexible as it allows to easily extend `events` type to handle more type of events in the future.

Reply:
> @sdavidp I don't see why subsequent runs could change those values affect the place and how we would store it. Can you elaborate more on that here or on slack?

We'd expect to be able to change the keys if they're not established in the beginning or for any other reason.

It should look like (for increasing snapshot incrementals,)

ordinal 10 - message:None - state: None
ordinal 11 - message:Signed(x) - state: Signed(x)
ordinal 12 - message:None - state: Signed(x)
ordinal 13 - message:Signed(y) - state: Signed(y)
ordinal 14 - message:None - state: Signed(y)